### DISCUSSION TITLE
Cannot find standard library headers of mingw gcc

### DISCUSSION THREAD

**Discussion by f-zl** [microsoft/vscode-cpptools#9260] [Q&A]
I've installed msvc and mingw64 gcc on my win10 machine, and only installed vscode extension "C/C++ Extension Pack" v1.1.0 with "C/C++" v1.9.8.

Cpp tools extension cannot find the standard library (say \<iostream\>) of mingw g++. But in quick fix, cpp tools suggest adding "D:/Program/msys64/mingw64/include/c++/11.2.0" or "D:/Program/msys64/mingw64/include/c++/11.2.0/tr1" in the includePath.

For C header <stdlib.h> in a .c file, cpp tools quick fix suggest adding "D:/Program/msys64/mingw64/include/c++/11.2.0/x86_64-w64-mingw32", which is the wrong path for the standard headers.

My c_cpp_properties.json is
> {
    "configurations": [
        {
            "name": "Win32",
            "includePath": [
                "${workspaceFolder}/**"
            ],
            "defines": [
                "_DEBUG",
                "UNICODE",
                "_UNICODE"
            ],
            "windowsSdkVersion": "10.0.19041.0",
            "cStandard": "c17",
            "cppStandard": "c++17",
            "intelliSenseMode": "windows-gcc-x64",
            "compilerPath": "D:/Program/msys64/mingw64/bin/gcc.exe"
        }
    ],
    "version": 4
}

What should I do to make cpp tools automatically include standard library headers of mingw gcc? Thanks very much!

**Comment by Colengms** [comment]
Hi @mjs225 .  Could you enable the setting `"C_Cpp.loggingLevel": "Debug"` and provide the output from the C/C++ output channel, leading up to the issue?  It sounds like the C/C++ extension may be encountering an issue trying to configure itself against the specified compiler.  The log output could contain some clues, such as error messages.

**Reply by f-zl** [reply]
> loggingLevel: Debug
cpptools/didChangeCppProperties
Attempting to get defaults from C compiler in "compilerPath" property: 'D:/Program/msys64/mingw64/bin/gcc.exe'
Querying compiler for default C++ language standard using command line: "D:/Program/msys64/mingw64/bin/gcc.exe" -x c++ -E -dM nul
Detected language standard version: gnu++17
Querying compiler for default C language standard using command line: "D:/Program/msys64/mingw64/bin/gcc.exe" -x c -E -dM nul
Detected language standard version: gnu17
Querying compiler's default target using command line: "D:/Program/msys64/mingw64/bin/gcc.exe" -dumpmachine
Compiler returned default target value: x86_64-w64-mingw32
Compiler query command line: "D:/Program/msys64/mingw64/bin/gcc.exe" -std=c17 -Wp,-v -E -dM -x c -m64  nul
Code browsing service initialized
Attempting to get defaults from C++ compiler in "compilerPath" property: 'D:/Program/msys64/mingw64/bin/gcc.exe'
Compiler query command line: "D:/Program/msys64/mingw64/bin/gcc.exe" -std=c++17 -Wp,-v -E -dM -x c++ -m64  nul
  Folder: D:/PROGRAM/MSYS64/MINGW64/INCLUDE/ will be indexed
  Folder: D:/TEST/VSCODE/MINGW/ will be indexed
Discovering files...
  Processing folder (recursive): D:/PROGRAM/MSYS64/MINGW64/INCLUDE/
  Processing folder (recursive): D:/TEST/VSCODE/MINGW/
  Discovering files: 25803 file(s) processed
  0 file(s) removed from database
Done discovering files.
Populating include completion cache.
Parsing remaining files...
  Parsing: 0 files(s) processed
Done parsing remaining files.
cpptools/getCodeActions: D:\test\vscode\mingw\hello.cpp (id: 2)
textDocument/didOpen: D:\test\vscode\mingw\hello.cpp
cpptools/textEditorSelectionChange
cpptools/getDocumentSymbols: D:\test\vscode\mingw\hello.cpp (id: 3)
cpptools/textEditorSelectionChange
cpptools/getSemanticTokens: D:\test\vscode\mingw\hello.cpp (id: 4)
cpptools/activeDocumentChange: D:\test\vscode\mingw\hello.cpp
cpptools/getDocumentSymbols
sending compilation args for D:\test\vscode\mingw\hello.cpp
  include: D:\PROGRAM\MSYS64\MINGW64\INCLUDE
  define: __DBL_MIN_EXP__=(-1021)
  define: __UINT_LEAST16_MAX__=0xffff
  define: __ATOMIC_ACQUIRE=2
  define: __FLT128_MAX_10_EXP__=4932
  define: __FLT_MIN__=1.17549435082228750796873653722224568e-38F
  define: __GCC_IEC_559_COMPLEX=2
  define: __UINT_LEAST8_TYPE__=unsigned char
  define: __SIZEOF_FLOAT80__=16
  define: __INTMAX_C(c)=c ## LL
  define: __CHAR_BIT__=8
  define: __MINGW32__=1
  define: __UINT8_MAX__=0xff
  define: __SCHAR_WIDTH__=8
  define: _WIN64=1
  define: __WINT_MAX__=0xffff
  define: __FLT32_MIN_EXP__=(-125)
  define: __ORDER_LITTLE_ENDIAN__=1234
  define: __SIZE_MAX__=0xffffffffffffffffULL
  define: __WCHAR_MAX__=0xffff
  define: __GCC_HAVE_SYNC_COMPARE_AND_SWAP_1=1
  define: __GCC_HAVE_SYNC_COMPARE_AND_SWAP_2=1
  define: __GCC_HAVE_SYNC_COMPARE_AND_SWAP_4=1
  define: __DBL_DENORM_MIN__=double(4.94065645841246544176568792868221372e-324L)
  define: __GCC_HAVE_SYNC_COMPARE_AND_SWAP_8=1
  define: __GCC_ATOMIC_CHAR_LOCK_FREE=2
  define: __GCC_IEC_559=2
  define: __FLT32X_DECIMAL_DIG__=17
  define: __FLT_EVAL_METHOD__=0
  define: __FLT64_DECIMAL_DIG__=17
  define: __GCC_ATOMIC_CHAR32_T_LOCK_FREE=2
  define: __UINT_FAST64_MAX__=0xffffffffffffffffULL
  define: __SIG_ATOMIC_TYPE__=int
  define: __DBL_MIN_10_EXP__=(-307)
  define: __FINITE_MATH_ONLY__=0
  define: __FLT32X_MAX_EXP__=1024
  define: __FLT32_HAS_DENORM__=1
  define: __UINT_FAST8_MAX__=0xff
  define: __DEC64_MAX_EXP__=385
  define: __INT8_C(c)=c
  define: __INT_LEAST8_WIDTH__=8
  define: __UINT_LEAST64_MAX__=0xffffffffffffffffULL
  define: __INT_LEAST8_MAX__=0x7f
  define: __SHRT_MAX__=0x7fff
  define: __LDBL_MAX__=1.18973149535723176502126385303097021e+4932L
  define: __FLT64X_MAX_10_EXP__=4932
  define: __LDBL_IS_IEC_60559__=2
  define: __FLT64X_HAS_QUIET_NAN__=1
  define: __UINT_LEAST8_MAX__=0xff
  define: __GCC_ATOMIC_BOOL_LOCK_FREE=2
  define: __FLT128_DENORM_MIN__=6.47517511943802511092443895822764655e-4966F128
  define: __UINTMAX_TYPE__=long long unsigned int
  define: __DEC32_EPSILON__=1E-6DF
  define: __FLT_EVAL_METHOD_TS_18661_3__=0
  define: __UINT32_MAX__=0xffffffffU
  define: __GXX_EXPERIMENTAL_CXX0X__=1
  define: __FLT128_MIN_EXP__=(-16381)
  define: __WINT_MIN__=0
  define: __FLT128_MIN_10_EXP__=(-4931)
  define: __FLT32X_IS_IEC_60559__=2
  define: __INT_LEAST16_WIDTH__=16
  define: __SCHAR_MAX__=0x7f
  define: __FLT128_MANT_DIG__=113
  define: __WCHAR_MIN__=0
  define: __INT64_C(c)=c ## LL
  define: __GCC_ATOMIC_POINTER_LOCK_FREE=2
  define: __FLT32X_MANT_DIG__=53
  define: __GCC_ATOMIC_CHAR16_T_LOCK_FREE=2
  define: __USER_LABEL_PREFIX__=
  define: __FLT32_MAX_10_EXP__=38
  define: __FLT64X_EPSILON__=1.08420217248550443400745280086994171e-19F64x
  define: __STDC_HOSTED__=1
  define: __DEC64_MIN_EXP__=(-382)
  define: __WIN64=1
  define: __DBL_DIG__=15
  define: __FLT32_DIG__=6
  define: __FLT_EPSILON__=1.19209289550781250000000000000000000e-7F
  define: __GXX_WEAK__=1
  define: __SHRT_WIDTH__=16
  define: __FLT32_IS_IEC_60559__=2
  define: __LDBL_MIN__=3.36210314311209350626267781732175260e-4932L
  define: __DBL_IS_IEC_60559__=2
  define: __DEC32_MAX__=9.999999E96DF
  define: __FLT64X_DENORM_MIN__=3.64519953188247460252840593361941982e-4951F64x
  define: __FLT32X_HAS_INFINITY__=1
  define: __INT32_MAX__=0x7fffffff
  define: __INT_WIDTH__=32
  define: __SIZEOF_LONG__=4
  define: __UINT16_C(c)=c
  define: __DECIMAL_DIG__=21
  define: __FLT64_EPSILON__=2.22044604925031308084726333618164062e-16F64
  define: __INT16_MAX__=0x7fff
  define: __FLT64_MIN_EXP__=(-1021)
  define: __FLT64X_MIN_10_EXP__=(-4931)
  define: __LDBL_HAS_QUIET_NAN__=1
  define: __FLT64_MANT_DIG__=53
  define: _REENTRANT=1
  define: __FLT64X_MANT_DIG__=64
  define: __GNUC__=11
  define: __GXX_RTTI=1
  define: __MMX__=1
  define: __FLT_HAS_DENORM__=1
  define: __SIZEOF_LONG_DOUBLE__=16
  define: __BIGGEST_ALIGNMENT__=16
  define: __STDC_UTF_16__=1
  define: __FLT64_MAX_10_EXP__=308
  define: __FLT32_HAS_INFINITY__=1
  define: __DBL_MAX__=double(1.79769313486231570814527423731704357e+308L)
  define: __INT_FAST32_MAX__=0x7fffffff
  define: __WINNT=1
  define: __DBL_HAS_INFINITY__=1
  define: __SIZEOF_FLOAT__=4
  define: __WINNT__=1
  define: __HAVE_SPECULATION_SAFE_VALUE=1
  define: __DEC32_MIN_EXP__=(-94)
  define: __INTPTR_WIDTH__=64
  define: __FLT64X_HAS_INFINITY__=1
  define: __UINT_LEAST32_MAX__=0xffffffffU
  define: __FLT32X_HAS_DENORM__=1
  define: __INT_FAST16_TYPE__=short int
  define: __STRICT_ANSI__=1
  define: __MMX_WITH_SSE__=1
  define: __LDBL_HAS_DENORM__=1
  define: __cplusplus=201703L
  define: __DEC32_MIN__=1E-95DF
  define: __DEPRECATED=1
  define: __DBL_MAX_EXP__=1024
  define: __WCHAR_WIDTH__=16
  define: __FLT32_MAX__=3.40282346638528859811704183484516925e+38F32
  define: __DEC128_EPSILON__=1E-33DL
  define: __SSE2_MATH__=1
  define: __ATOMIC_HLE_RELEASE=131072
  define: __WIN32__=1
  define: __PTRDIFF_MAX__=0x7fffffffffffffffLL
  define: __amd64=1
  define: __ATOMIC_HLE_ACQUIRE=65536
  define: __GNUG__=11
  define: __LONG_LONG_MAX__=0x7fffffffffffffffLL
  define: __SIZEOF_SIZE_T__=8
  define: __FLT64X_MIN_EXP__=(-16381)
  define: __SIZEOF_WINT_T__=2
  define: __LONG_LONG_WIDTH__=64
  define: __FLT32_MAX_EXP__=128
  define: __GXX_ABI_VERSION=1016
  define: __FLT128_HAS_INFINITY__=1
  define: __FLT_MIN_EXP__=(-125)
  define: __x86_64=1
  define: __INT_FAST64_TYPE__=long long int
  define: __FLT64_DENORM_MIN__=4.94065645841246544176568792868221372e-324F64
  define: __DBL_MIN__=double(2.22507385850720138309023271733240406e-308L)
  define: __FLT128_EPSILON__=1.92592994438723585305597794258492732e-34F128
  define: __FLT64X_NORM_MAX__=1.18973149535723176502126385303097021e+4932F64x
  define: __SIZEOF_POINTER__=8
  define: __SIZE_TYPE__=long long unsigned int
  define: __DBL_HAS_QUIET_NAN__=1
  define: __FLT32X_EPSILON__=2.22044604925031308084726333618164062e-16F32x
  define: __DECIMAL_BID_FORMAT__=1
  define: __GXX_TYPEINFO_EQUALITY_INLINE=0
  define: __FLT64_MIN_10_EXP__=(-307)
  define: __FLT64X_DECIMAL_DIG__=21
  define: __DEC128_MIN__=1E-6143DL
  define: __REGISTER_PREFIX__=
  define: __UINT16_MAX__=0xffff
  define: __cdecl=__attribute__((__cdecl__))
  define: __LDBL_HAS_INFINITY__=1
  define: __FLT32_MIN__=1.17549435082228750796873653722224568e-38F32
  define: __UINT8_TYPE__=unsigned char
  define: __FLT_DIG__=6
  define: __NO_INLINE__=1
  define: __DEC_EVAL_METHOD__=2
  define: __DEC128_MAX__=9.999999999999999999999999999999999E6144DL
  define: __FLT_MANT_DIG__=24
  define: __LDBL_DECIMAL_DIG__=21
  define: __VERSION__="11.2.0"
  define: __UINT64_C(c)=c ## ULL
  define: _WIN32=1
  define: __SEH__=1
  define: __INT_LEAST32_MAX__=0x7fffffff
  define: __GCC_ATOMIC_INT_LOCK_FREE=2
  define: __FLT128_MAX_EXP__=16384
  define: __FLT32_MANT_DIG__=24
  define: __FLOAT_WORD_ORDER__=__ORDER_LITTLE_ENDIAN__
  define: __FLT128_HAS_DENORM__=1
  define: __FLT32_DECIMAL_DIG__=9
  define: __FLT128_DIG__=33
  define: __INT32_C(c)=c
  define: __DEC64_EPSILON__=1E-15DD
  define: __ORDER_PDP_ENDIAN__=3412
  define: __DEC128_MIN_EXP__=(-6142)
  define: __INT_FAST32_TYPE__=int
  define: __UINT_LEAST16_TYPE__=short unsigned int
  define: __DBL_HAS_DENORM__=1
  define: __UINT64_MAX__=0xffffffffffffffffULL
  define: __FLT_IS_IEC_60559__=2
  define: __GNUC_WIDE_EXECUTION_CHARSET_NAME="UTF-16LE"
  define: __FLT64X_DIG__=18
  define: __INT8_TYPE__=signed char
  define: __GCC_ASM_FLAG_OUTPUTS__=1
  define: __UINT32_TYPE__=unsigned int
  define: __FLT_RADIX__=2
  define: __INT_LEAST16_TYPE__=short int
  define: __LDBL_EPSILON__=1.08420217248550443400745280086994171e-19L
  define: __UINTMAX_C(c)=c ## ULL
  define: __k8=1
  define: __FLT32X_MIN__=2.22507385850720138309023271733240406e-308F32x
  define: __SIG_ATOMIC_MAX__=0x7fffffff
  define: __GCC_ATOMIC_WCHAR_T_LOCK_FREE=2
  define: __SIZEOF_PTRDIFF_T__=8
  define: __LDBL_DIG__=18
  define: __FLT64_IS_IEC_60559__=2
  define: __x86_64__=1
  define: __FLT32X_MIN_EXP__=(-1021)
  define: __DEC32_SUBNORMAL_MIN__=0.000001E-95DF
  define: __MSVCRT__=1
  define: __INT_FAST16_MAX__=0x7fff
  define: __FLT64_DIG__=15
  define: __UINT_FAST32_MAX__=0xffffffffU
  define: __UINT_LEAST64_TYPE__=long long unsigned int
  define: __FLT_HAS_QUIET_NAN__=1
  define: __FLT_MAX_10_EXP__=38
  define: __LONG_MAX__=0x7fffffffL
  define: __FLT64X_HAS_DENORM__=1
  define: __DEC128_SUBNORMAL_MIN__=0.000000000000000000000000000000001E-6143DL
  define: __FLT_HAS_INFINITY__=1
  define: __GNUC_EXECUTION_CHARSET_NAME="UTF-8"
  define: __UINT_FAST16_TYPE__=short unsigned int
  define: __DEC64_MAX__=9.999999999999999E384DD
  define: __INT_FAST32_WIDTH__=32
  define: __CHAR16_TYPE__=short unsigned int
  define: __PRAGMA_REDEFINE_EXTNAME=1
  define: __SIZE_WIDTH__=64
  define: __SEG_FS=1
  define: __INT_LEAST16_MAX__=0x7fff
  define: __DEC64_MANT_DIG__=16
  define: __INT64_MAX__=0x7fffffffffffffffLL
  define: __SEG_GS=1
  define: __FLT32_DENORM_MIN__=1.40129846432481707092372958328991613e-45F32
  define: __SIG_ATOMIC_WIDTH__=32
  define: __INT_LEAST64_TYPE__=long long int
  define: __INT16_TYPE__=short int
  define: __INT_LEAST8_TYPE__=signed char
  define: __SIZEOF_INT__=4
  define: __DEC32_MAX_EXP__=97
  define: __INT_FAST8_MAX__=0x7f
  define: __FLT128_MAX__=1.18973149535723176508575932662800702e+4932F128
  define: __INTPTR_MAX__=0x7fffffffffffffffLL
  define: __FLT64_HAS_QUIET_NAN__=1
  define: __stdcall=__attribute__((__stdcall__))
  define: __FLT32_MIN_10_EXP__=(-37)
  define: __EXCEPTIONS=1
  define: __GXX_MERGED_TYPEINFO_NAMES=0
  define: __PTRDIFF_WIDTH__=64
  define: __LDBL_MANT_DIG__=64
  define: __FLT64_HAS_INFINITY__=1
  define: __FLT64X_MAX__=1.18973149535723176502126385303097021e+4932F64x
  define: __STDCPP_DEFAULT_NEW_ALIGNMENT__=16
  define: __SIG_ATOMIC_MIN__=(-__SIG_ATOMIC_MAX__ - 1)
  define: __GCC_ATOMIC_LONG_LOCK_FREE=2
  define: __DEC32_MANT_DIG__=7
  define: __k8__=1
  define: __INTPTR_TYPE__=long long int
  define: __UINT16_TYPE__=short unsigned int
  define: __WCHAR_TYPE__=short unsigned int
  define: __pic__=1
  define: __UINTPTR_MAX__=0xffffffffffffffffULL
  define: __INT_FAST64_WIDTH__=64
  define: __INT_FAST64_MAX__=0x7fffffffffffffffLL
  define: __GCC_ATOMIC_TEST_AND_SET_TRUEVAL=1
  define: __FLT_NORM_MAX__=3.40282346638528859811704183484516925e+38F
  define: __FLT64X_MAX_EXP__=16384
  define: __UINT_FAST64_TYPE__=long long unsigned int
  define: __INT_MAX__=0x7fffffff
  define: __code_model_medium__=1
  define: __INT64_TYPE__=long long int
  define: __FLT_MAX_EXP__=128
  define: __ORDER_BIG_ENDIAN__=4321
  define: __DBL_MANT_DIG__=53
  define: __SIZEOF_FLOAT128__=16
  define: __INT_LEAST64_MAX__=0x7fffffffffffffffLL
  define: __DEC64_MIN__=1E-383DD
  define: __WINT_TYPE__=short unsigned int
  define: __UINT_LEAST32_TYPE__=unsigned int
  define: __SIZEOF_SHORT__=2
  define: __FLT32_NORM_MAX__=3.40282346638528859811704183484516925e+38F32
  define: __SSE__=1
  define: __LDBL_MIN_EXP__=(-16381)
  define: __FLT64_MAX__=1.79769313486231570814527423731704357e+308F64
  define: __amd64__=1
  define: __WINT_WIDTH__=16
  define: __INT_LEAST64_WIDTH__=64
  define: __LDBL_MAX_EXP__=16384
  define: __FLT32X_MAX_10_EXP__=308
  define: __WIN32=1
  define: __SIZEOF_INT128__=16
  define: __FLT64X_IS_IEC_60559__=2
  define: __WCHAR_UNSIGNED__=1
  define: __LDBL_MAX_10_EXP__=4932
  define: __ATOMIC_RELAXED=0
  define: __DBL_EPSILON__=double(2.22044604925031308084726333618164062e-16L)
  define: __thiscall=__attribute__((__thiscall__))
  define: __FLT128_MIN__=3.36210314311209350626267781732175260e-4932F128
  define: __UINT8_C(c)=c
  define: __FLT64_MAX_EXP__=1024
  define: __INT_LEAST32_TYPE__=int
  define: __SIZEOF_WCHAR_T__=2
  define: __GNUC_PATCHLEVEL__=0
  define: __FLT128_NORM_MAX__=1.18973149535723176508575932662800702e+4932F128
  define: __FLT64_NORM_MAX__=1.79769313486231570814527423731704357e+308F64
  define: __FLT128_HAS_QUIET_NAN__=1
  define: __INTMAX_MAX__=0x7fffffffffffffffLL
  define: __INT_FAST8_TYPE__=signed char
  define: __fastcall=__attribute__((__fastcall__))
  define: __FLT64X_MIN__=3.36210314311209350626267781732175260e-4932F64x
  define: __STDCPP_THREADS__=1
  define: __GNUC_STDC_INLINE__=1
  define: __FLT64_HAS_DENORM__=1
  define: __FLT32_EPSILON__=1.19209289550781250000000000000000000e-7F32
  define: __DBL_DECIMAL_DIG__=17
  define: __STDC_UTF_32__=1
  define: __INT_FAST8_WIDTH__=8
  define: __FXSR__=1
  define: __FLT32X_MAX__=1.79769313486231570814527423731704357e+308F32x
  define: __DBL_NORM_MAX__=double(1.79769313486231570814527423731704357e+308L)
  define: __BYTE_ORDER__=__ORDER_LITTLE_ENDIAN__
  define: __MINGW64__=1
  define: __INTMAX_WIDTH__=64
  define: __UINT64_TYPE__=long long unsigned int
  define: __UINT32_C(c)=c ## U
  define: __FLT_DENORM_MIN__=1.40129846432481707092372958328991613e-45F
  define: __FLT128_IS_IEC_60559__=2
  define: __INT8_MAX__=0x7f
  define: __LONG_WIDTH__=32
  define: __PIC__=1
  define: __UINT_FAST32_TYPE__=unsigned int
  define: __FLT32X_NORM_MAX__=1.79769313486231570814527423731704357e+308F32x
  define: __CHAR32_TYPE__=unsigned int
  define: __FLT_MAX__=3.40282346638528859811704183484516925e+38F
  define: __SSE2__=1
  define: __INT32_TYPE__=int
  define: __SIZEOF_DOUBLE__=8
  define: __FLT_MIN_10_EXP__=(-37)
  define: __FLT64_MIN__=2.22507385850720138309023271733240406e-308F64
  define: __INT_LEAST32_WIDTH__=32
  define: __INTMAX_TYPE__=long long int
  define: _INTEGRAL_MAX_BITS=64
  define: __DEC128_MAX_EXP__=6145
  define: __FLT32X_HAS_QUIET_NAN__=1
  define: __ATOMIC_CONSUME=1
  define: __GNUC_MINOR__=2
  define: __INT_FAST16_WIDTH__=16
  define: __UINTMAX_MAX__=0xffffffffffffffffULL
  define: __FLT32X_DENORM_MIN__=4.94065645841246544176568792868221372e-324F32x
  define: __DBL_MAX_10_EXP__=308
  define: __LDBL_DENORM_MIN__=3.64519953188247460252840593361941982e-4951L
  define: __INT16_C(c)=c
  define: __STDC__=1
  define: __FLT32X_DIG__=15
  define: __PTRDIFF_TYPE__=long long int
  define: __ATOMIC_SEQ_CST=5
  define: __FLT32X_MIN_10_EXP__=(-307)
  define: __UINTPTR_TYPE__=long long unsigned int
  define: __DEC64_SUBNORMAL_MIN__=0.000000000000001E-383DD
  define: __DEC128_MANT_DIG__=34
  define: __LDBL_MIN_10_EXP__=(-4931)
  define: __SSE_MATH__=1
  define: __SIZEOF_LONG_LONG__=8
  define: __FLT128_DECIMAL_DIG__=36
  define: __GCC_ATOMIC_LLONG_LOCK_FREE=2
  define: __FLT32_HAS_QUIET_NAN__=1
  define: __FLT_DECIMAL_DIG__=9
  define: __UINT_FAST16_MAX__=0xffff
  define: __LDBL_NORM_MAX__=1.18973149535723176502126385303097021e+4932L
  define: __GCC_ATOMIC_SHORT_LOCK_FREE=2
  define: __UINT_FAST8_TYPE__=unsigned char
  define: __WIN64__=1
  define: __ATOMIC_ACQ_REL=4
  define: __ATOMIC_RELEASE=3
  define: __declspec(x)=__attribute__((x))
  define: _DEBUG
  define: UNICODE
  define: _UNICODE
  other: --g++
  other: --gnu_version=110200
  stdver: c++17
  intelliSenseMode: windows-gcc-x64
Checking for syntax errors: D:\test\vscode\mingw\hello.cpp
Queueing IntelliSense update for files in translation unit of: D:\test\vscode\mingw\hello.cpp
cpptools/finishUpdateSquiggles
Error squiggle count: 15
Error squiggles will be disabled in: file:///D%3A/test/vscode/mingw/hello.cpp
Update IntelliSense time (sec): 1.1
cpptools/getCodeActions: D:\test\vscode\mingw\hello.cpp (id: 5)
cpptools/getFoldingRanges: D:\test\vscode\mingw\hello.cpp (id: 6)
cpptools/getCodeActions: D:\test\vscode\mingw\hello.cpp (id: 7)
cpptools/getFoldingRanges: D:\test\vscode\mingw\hello.cpp (id: 8)

The hello.cpp is
```cpp
#include <iostream>
#include <vector>
#include <string>

using namespace std;

int main()
{
    vector<string> msg {"Hello", "C++", "World", "from", "VS Code", "and the C++ extension!"};

    for (const string& word : msg)
    {
        cout << word << " ";
    }
    cout << endl;
}
```

**Reply by Colengms** [reply]
Hi @mjs225 .  It looks like the extension has successfully queried the compiler, and provided an IntelliSense configuration to our IntelliSense parser that includes compiler-specific defines and includes.  The only include path provided by the compiler is:

```
D:\PROGRAM\MSYS64\MINGW64\INCLUDE
```

However, my own local copy of MinGW returns the following paths:

```
  include: C:\MSYS64\MINGW64\INCLUDE\C++\11.2.0
  include: C:\MSYS64\MINGW64\INCLUDE\C++\11.2.0\X86_64-W64-MINGW32
  include: C:\MSYS64\MINGW64\INCLUDE\C++\11.2.0\BACKWARD
  include: C:\MSYS64\MINGW64\LIB\GCC\X86_64-W64-MINGW32\11.2.0\INCLUDE
  include: C:\MSYS64\MINGW64\INCLUDE
  include: C:\MSYS64\MINGW64\LIB\GCC\X86_64-W64-MINGW32\11.2.0\INCLUDE-FIXED
  include: C:\MSYS64\MINGW64\X86_64-W64-MINGW32\INCLUDE
  include: Z:\MINGW64\INCLUDE
```

Running the equivalent of the following command against my own install of MinGW results in display of those paths.  What does the following display when you run the command directly?  (Note: it's likely you will need to try this command with the current working directory set to the same bin folder, or it may fail to locate certain files it needs)

```
"D:/Program/msys64/mingw64/bin/gcc.exe" -std=c++17 -Wp,-v -E -x c++ -m64 nul
```

If the full set of system include paths are not present, there would seem to be an issue with your installation of MinGW.  Is it possible you have not install the C++ (g++, etc) components of MinGW?

**Reply by f-zl** [reply]
Running ```"D:/Program/msys64/mingw64/bin/gcc.exe" -std=c++17 -Wp,-v -E -x c++ -m64 nul``` in cmd.exe gives this:
> ignoring duplicate directory "D:/Program/msys64/mingw64/lib/gcc/../../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../include/c++/11.2.0"
ignoring duplicate directory "D:/Program/msys64/mingw64/lib/gcc/../../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../include/c++/11.2.0/x86_64-w64-mingw32"
ignoring duplicate directory "D:/Program/msys64/mingw64/lib/gcc/../../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../include/c++/11.2.0/backward"
ignoring duplicate directory "D:/Program/msys64/mingw64/lib/gcc/../../lib/gcc/x86_64-w64-mingw32/11.2.0/include"
ignoring nonexistent directory "D:/a/msys64/mingw64/include"
ignoring nonexistent directory "/mingw64/include"
ignoring duplicate directory "D:/Program/msys64/mingw64/lib/gcc/../../lib/gcc/x86_64-w64-mingw32/11.2.0/include-fixed"
ignoring duplicate directory "D:/Program/msys64/mingw64/lib/gcc/../../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../x86_64-w64-mingw32/include"
ignoring nonexistent directory "D:/a/msys64/mingw64/x86_64-w64-mingw32/include"
#include "..." search starts here:
#include <...> search starts here:
 D:/Program/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../include/c++/11.2.0
 D:/Program/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../include/c++/11.2.0/x86_64-w64-mingw32
 D:/Program/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../include/c++/11.2.0/backward
 D:/Program/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/include
 D:/Program/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../include
 D:/Program/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/include-fixed
 D:/Program/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../x86_64-w64-mingw32/include
End of search list.
\# 0 "nul"
\# 0 "<built-in>"
\# 0 "<command-line>"
\# 1 "nul"

P.S. ```D:\Program\msys64\mingw64\bin\``` is in the PATH and ```where gcc``` gives the right result.

**Reply by f-zl** [reply]
I've found it's the "C_Cpp.default.systemIncludePath" in the User's settings that cause the problem. At some point, I added "D:\\Program\\msys64\\mingw64\\include" to try solving the missing header problem. After removing that, the problem disappeared. Anyway, thanks very much for your help!
